### PR TITLE
Upload build-log.txt if it exists.

### PR DESCRIPTION
**Pull request by spxtr**
Companion to https://github.com/kubernetes/test-infra/pull/112.

### CODE REVIEW

File: hack/jenkins/upload-to-gcs.sh [hunk 1]
```diff
@@ -111,6 +111,10 @@ function upload_artifacts_and_build_result() {
       gsutil -m -q -o "GSUtil:use_magicfile=True" cp -a "${gcs_acl}" -r -c \
         -z log,txt,xml "${artifacts_path}" "${gcs_build_path}/artifacts" || continue
     fi
+    if [[ -e "${WORKSPACE}/build-log.txt" ]]; then
+      echo "Uploading build log"
+      gsutil -q cp -a "${gcs_acl}" "${WORKSPACE}/build-log.txt" "${gcs_build_path}"
```
Comment (ixdy) [inline hack/jenkins/upload-to-gcs.sh]: maybe add `-z txt` so this gets compressed?

Comment (ixdy) [inline hack/jenkins/upload-to-gcs.sh]: or even just `-Z`

Comment (spxtr) [inline hack/jenkins/upload-to-gcs.sh]: Done.

**Comment by ixdy** [comment]
LGTM

**Comment by ixdy** [comment]
I think SQ is only merging things for 1.3 right now.

**Comment by spxtr** [comment]
Oh, right. I think it's okay for this to go in right away.

**Pull request merged**